python-threatexchange/threatexchange/exchanges/signal_exchange_api.py

```diff
@@ -117,23 +105,23 @@ def get_name(cls) -> str:
                 name = name[: -len(suffix)]
         return common.class_name_to_human_name(name, "Signal")
 
-    @classmethod
-    def get_checkpoint_cls(cls) -> t.Type[state.TFetchCheckpoint]:
+    @staticmethod
+    @abstractmethod
+    def get_checkpoint_cls() -> t.Type[state.TFetchCheckpoint]:
         """Returns the dataclass used to control checkpoint for this API"""
-        # Default = no checkpoints
-        return state.FetchCheckpointBase  # type: ignore
+        pass
 
-    @classmethod
-    def get_record_cls(cls) -> t.Type[state.TFetchedSignalMetadata]:
+    @staticmethod
+    @abstractmethod
+    def get_record_cls() -> t.Type[state.TFetchedSignalMetadata]:
         """Returns the dataclass used to store records for this API"""
-        # Default = no metadata
-        return state.FetchedSignalMetadata  # type: ignore
+        pass
 
-    @classmethod
-    def get_config_class(cls) -> t.Type[TCollabConfig]:
+    @staticmethod
+    @abstractmethod
+    def get_config_cls() -> t.Type[TCollabConfig]:
         """Returns the dataclass used to store records for this API"""
-        # Default - just knowing the type is enough
-        return CollaborationConfigBase  # type: ignore
+        pass
```

Comment:
It's expected that these are hardcoded, and not able to have any logic. Since `cls` should be unused in every case, and classmethod/staticmethod are equivalent in heritability (see above), seems more correct to static